<?xml version="1.0"?>
<robot name="my_robot" xmlns:xacro="http://www.ros.org/wiki/xacro">

    <!-- Base Link -->
    <link name="base_link">
    </link>

    <!-- BASE_FOOTPRINT LINK -->

    <joint name="base_footprint_joint" type="fixed">
        <parent link="base_link"/>
        <child link="base_footprint"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>

    <link name="base_footprint">
    </link>

    <!-- Chassis -->
    <joint name="chassis_joint" type="fixed">
        <parent link="base_link"/>
        <child link="chassis"/>
        <origin xyz="0.0 0 -0.01"/>
    </joint>
    <link name="chassis">
        <visual>
            <origin xyz="0.0 0 0.055"/>
            <geometry>
                <box size="0.80 0.58 0.20"/>
            </geometry>
            <material name="orange">
                <color rgba="1 0.3 0.1 1"/>
            </material>
        </visual>
        <collision>
            <origin xyz="0.15 0 0.055"/>
            <geometry>
                <box size="0.30 0.16 0.11"/>
            </geometry>
        </collision>
        <inertial>
            <mass value="1.0"/>
            <origin xyz="0.15 0 0.055" rpy="0 0 0"/>
            <inertia ixx="0.0011" ixy="0" ixz="0" iyy="0.0011" iyz="0" izz="0.0011"/>
        </inertial>
    </link>

    <!-- Left Wheel -->
    <joint name="left_wheel_joint" type="continuous">
        <parent link="base_link"/>
        <child link="left_wheel"/>
        <origin xyz="0 0.29 0" rpy="-1.5708 0 0" />
        <axis xyz="0 0 1"/>
    </joint>
    <link name="left_wheel">
        <visual>
            <geometry>
                <cylinder radius="0.1" length="0.04"/>
            </geometry>
            <material name="blue">
                <color rgba="0.2 0.2 1 1"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <cylinder radius="0.01" length="0.04"/>
            </geometry>
        </collision>
        <inertial>
            <mass value="0.05"/>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <inertia ixx="0.00001" ixy="0" ixz="0" iyy="0.00001" iyz="0" izz="0.00001"/>
        </inertial>
    </link>

    <!-- Right Wheel -->
    <joint name="right_wheel_joint" type="continuous">
        <parent link="base_link"/>
        <child link="right_wheel"/>
        <origin xyz="0 -0.29 0" rpy="1.5708 0 0" />
        <axis xyz="0 0 -1"/>
    </joint>
    <link name="right_wheel">
        <visual>
            <geometry>
                <cylinder radius="0.1" length="0.04"/>
            </geometry>
            <material name="blue">
                <color rgba="0.2 0.2 1 1"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <cylinder radius="0.1" length="0.04"/>
            </geometry>
        </collision>
        <inertial>
            <mass value="0.05"/>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <inertia ixx="0.00001" ixy="0" ixz="0" iyy="0.00001" iyz="0" izz="0.00001"/>
        </inertial>
    </link>

    <!-- Laser Joint -->
    <joint name="laser_joint" type="fixed">
        <parent link="base_link"/>
        <child link="laser_frame"/>
        <origin xyz="0.33 0 0.20" rpy="0 0 3.14159"/>
    </joint>

    <!-- Laser Frame -->
    <link name="laser_frame">
        <visual>
            <geometry>
                <cylinder radius="0.05" length="0.04"/>
            </geometry>
            <material name="black">
                <color rgba="0.0 0.0 0.0 1.0"/>
            </material>
        </visual>
        <visual>
            <origin xyz="0 0 -0.05"/>
            <geometry>
                <cylinder radius="0.01" length="0.1"/>
            </geometry>
            <material name="black">
                <color rgba="0.0 0.0 0.0 1.0"/>
            </material>
        </visual>
    </link>
    <!-- IMU -->
    <!-- <joint name="imu_joint" type="fixed">
        <parent link="chassis"/>
        <child link="imu_link"/>
        <origin xyz="0.15 0 0.055" rpy="0 0 0"/>
    </joint>
    <link name="imu_link">
        <visual>
            <geometry>
                <box size="0.05 0.05 0.05"/>
            </geometry>
            <material name="red">
                <color rgba="1 0 0 1"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <box size="0.05 0.05 0.05"/> -->
            <!-- </geometry>
        </collision>
        <inertial>
            <mass value="0.01"/>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <inertia ixx="0.0001" ixy="0" ixz="0" iyy="0.0001" iyz="0" izz="0.0001"/>
        </inertial>
    </link> -->

</robot>
